src/users/index.ts

```diff
@@ -0,0 +1,65 @@
+import jsonwebtoken from "jsonwebtoken"
+import { requestGet, requestPost, requestDelete } from "../entityRequest"
+import {
+  ICredentials,
+  IEnvironment,
+  IFavorite,
+  IFavoritesResource,
+  IFetchExecutor
+} from "../types"
+
+const users = (
+  creds: ICredentials,
+  Ienvironment?: IEnvironment,
+  fetchExecutor?: IFetchExecutor
+): IFavoritesResource<IFavorite> => ({
+  addFavorite: (id: number, type: "Community" | "MobileHome") => {
+    const decoded: any = jsonwebtoken.decode(creds.token ? creds.token : "")
+    if (!decoded) {
+      return new Promise(resolve => {
+        resolve()
+      })
+    }
+    const body =
+      type === "Community" ? { community_id: id } : { mobilehome_id: id }
+    return requestPost<IFavorite>(
+      `v1/users/${decoded.user_id}/favorites`,
+      creds,
+      "Favorites",
+      Ienvironment,
+      fetchExecutor,
+      body
+    )
+  },
+  deleteFavorite: (id: number) => {
```

Comment:
I added the ability for administrators to update another user's data and checked that they are authorized when the call is made. Can we make the user_id a parameter that is passed in instead of pulled from the token. The React client will do this decoding of the token and pass it to the API. When we make an OpenAPI definition for the API, generated code will create a method with 2 parameters anyway, so it will be easier to follow along across programming languages.

What do you think about having a method in mhbo-js that clients use to extract the user_id? So instead of the React app having to know specifically that the field name is "user_id" in the token, you can just call getUserId(jsonwebtoken). That way if we change the field or modify defaults we can do it in one place. We could just place it in users.getUserId or maybe put it in a Token utility of some sort. What do you think?

Reply:
Added a token utility to get userId along with IToken type so that other information can be extracted from the token later.